ui/plugin-system/src/components/DatasourceSelect.tsx

```diff
@@ -32,51 +33,124 @@ export interface DatasourceSelectProps extends Omit<SelectProps<string>, Omitted
  */
 export function DatasourceSelect(props: DatasourceSelectProps) {
   const { datasourcePluginKind, value, onChange, ...others } = props;
-  const { data, isLoading } = useListDatasources(datasourcePluginKind);
+  const { data, isLoading } = useListDatasourceSelectItems(datasourcePluginKind);
 
-  // Convert the datasource list into menu items with name/value strings that the Select input can work with
+  // Rebuild the group of the value if not provided
+  const defaultValue = useMemo(() => {
```

Comment:
Note to myself : this is still not fully working. In some cases, nothing is selected. I think it is when there is no datasource selected and no default datasource exist in the items list. 

This specific case could be encountered and should be more explicit to the user. 

Then if it occurs
- should we select the first of the list as default?
- should we put a warning (error?) saying that if you don't explicitly select a datasource, none of them are declared as default?

Reply:
For the record, we discussed together and here is the output: 
If no default datasource exist, take the first one, but it should still be considered as default (no ds selected in the backend). This is possible using the same way the store provider setup the classic default. It means that there will always be a default value in the selection list.